source/SH3/graphics/texture.cpp

```diff
@@ -4,44 +4,62 @@
  *
  *  @copyright 2016  Palm Studios
  *
- *
  *  @date 2-1-2017
  *
  *  @author Jesse Buhagiar
  */
 #include <SH3/graphics/texture.hpp>
 #include <SH3/system/log.hpp>
+#include <SH3/arc/vfile.hpp>
+
+#include <cassert>
+
 
 using namespace sh3_graphics;
 
-std::string texture::load_error::message() const
+//TODO: Scale the texture and then
+void sh3_texture::Load(sh3_arc& mft, const std::string& filename)
 {
-    std::string error;
-    switch(result)
-    {
-    case load_result::SUCCESS:
-        error = "Success";
-        break;
-    case load_result::INVALID_CHUNK:
-        error = "Invalid chunk";
-        break;
-    case load_result::END_OF_FILE:
-        error = "End of file";
-        break;
-    }
-    return error;
-};
-
-texture::load_error sh3_graphics::texture::Load(const std::string& filename, sh3_arc& arc)
-{
-    load_error error;
+    sh3_texture_header header;
+    sh3_arc_vfile file(mft, filename);
+    sh3_arc_vfile::read_error e;
+
+    int ret = file.ReadData(&header, sizeof(sh3_texture_header), e); // Read in the header
+
+    if(ret != sizeof(sh3_texture_header))
+        die("sh3_texture::Load( ): Header only read %d bytes of data! It should have been %d", ret, sizeof(sh3_texture_header)); // TODO: Fallback on dummy texture (like Source's purple and black checkboard)
+
+    if(header.bpp == 8)
+        Log(LogLevel::WARN, "WARNING: 8-bit textures are currently unsupported! They WILL look like nothing but pixel garbage.");
+
+    Log(LogLevel::INFO, "Texture info:\n\ttexWidth:%d\n\ttexHeight:%d\n\ttexSize:%d\n\tbpp:%d", header.texWidth, header.texHeight, header.texSize, header.bpp);
+
+    file.Seek(sizeof(sh3_texture_header) + header.dataOffset, std::ios_base::beg);
 
-    if(arc.LoadFile(filename, pixbuff) == ARC_FILE_NOT_FOUND)
-    {
-        die("sh3_graphics::texture::Load( ): Unable to find file %s!", filename.c_str());
-    }
+    data.resize(header.texSize);
+    ret = file.ReadData(&data[0], header.texSize, e);
 
-    error.set_error(load_result::END_OF_FILE);
+    if(ret != header.texSize)
+        Log(LogLevel::WARN, "sh3_texture::Load( ): Only read %d out of %d bytes!", ret, header.texSize);
 
-    return error;
+    // Now actually create a logical texture with OpenGL on the gpu
+    glGenTextures(1, &tex);
+
+    glBindTexture(GL_TEXTURE_2D, tex);
+    glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, header.texWidth, header.texHeight, 0, GL_BGRA, GL_UNSIGNED_BYTE, &data[0]);
+    glBindTexture(GL_TEXTURE_2D, 0); // Unbind this texture
```

Comment:
Yeah they have to be to be drawn. Basically (how I'm thinking it'll work), is each mesh has a texture (*.tex). Everything is loaded before the scene is rendered. During the render loop, we transverse a list containing each mesh. Its VBO and texture is bound and drawn etc. 

When you generate a texture I'm pretty sure it's recommended you unbind it. Otherwise you could add a bunch of stuff to another texture without realising it.